### DISCUSSION TITLE
Hitdef sound channels [feature request]

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#913] [Feature Request] [answered]
While I'm not in a hurry to request new features, I think this one should be easy to implement and have a positive impact in the engine.

Hitdef could use two new parameters: "hitsound.channel" and "guardsound.channel". They should be self-explanatory. They could default to -1, which would be whatever channel is available, like the current behavior (and Mugen's).

Alternatively, this could be a constant instead (still defaulting to -1).

This is one of those things I and a few others had to code manually to make the characters feel more like they're out of a fighting game. It's subtle, but chaotic hit sound channels is one of the things that makes Mugen feel like Mugen (in a bad way). Most fighting games have each hitsound interrupting the last.

It'd be even better if all characters could just play these sounds in a common channel (like a screenpack channel or something like that), but the previous suggestion would be more than good enough.

**Comment by Sluigi123** [comment]
I agree with this idea! Implementing each of the [hit/guard] sounds to play over each other by hand uses up more lines of code than necessary. Reducing them down to parameters also improves the overall organization as well. ;)

**Comment by K4thos** [comment, marked as answer]
Feature request has been fulfilled in this commit: https://github.com/ikemen-engine/Ikemen-GO/commit/2ee7b036e452875c7ee41a4f94619ce88a79d660
I would appreciate if you guys test it and let me know if there are any problems.
New functionality:
- Hidef/Projectile hitsound.channel and guardsound.channel parameters (defaults to -1)
- Helper InheritChannels parameters (works similarly to InheritJuggles: 1 makes helper to use parent's channels, if parent exists, 2 makes helper use root channels)
